api/v1alpha1/timeout_types.go

```diff
@@ -41,3 +41,18 @@ type HTTPTimeout struct {
 	// +optional
 	MaxConnectionDuration *gwapiv1.Duration `json:"maxConnectionDuration,omitempty"`
 }
+
+type ClientTimeout struct {
+	// Timeout settings for HTTP.
+	//
+	// +optional
+	HTTP *HTTPClientTimeout `json:"http,omitempty"`
+}
+
+type HTTPClientTimeout struct {
+	// The timeout for processing the client request. Starting upon request initiation and ending when the last byte is sent to backend.
+	// Default: 300 seconds.
+	//
+	// +optional
+	RequestProcessTimeout *gwapiv1.Duration `json:"requestProcessTimeout,omitempty"`
```

Comment:
`requestProceess` doesnt convey the exact meaning of the knob imo
thoughts on `processingRequestTimeout` or even `requestTimeout` ?

Reply:
+1 for `requestTimeout` with a descriptive comment indicating something closer to what the docs describe:

> The amount of time that Envoy will wait for the entire request to be received. The timer is activated when the request is initiated, and is disarmed when the last byte of the request is sent upstream OR when the response is initiated.

Alternately something like `requestRecievedTimeout` seems less ambiguous than any variation of `process` timeout and conveys the intent while differentiating this from the (upstream) request timeout which can be specified via timeouts.request on the HTTPRoute

Reply:
+1 for `requestRecievedTimeout`

Reply:
`requestTimeout` is slightly overloaded as gateway-api already defines a structure like `timeout.request` for [httproutes](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteTimeouts). 

I'm  fine with `requestRecievedTimeout` or just `requestTimeout` if we're not concerned that this would be confusing (due to the other GW API field).

Reply:
Thanks for the feedback, I changed the field name to `requestRecievedTimeout` and added a more descriptive comment as @davidalger suggested.